packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/editingcontext/eventprocessorregistry/EditingContextEventProcessorRegistryIntegrationTests.java

```diff
@@ -0,0 +1,91 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Integration tests of the {@link EditingContextEventProcessorRegistry}.
+ *
+ * @author arichard
+ */
+@Transactional
+@SuppressWarnings("checkstyle:MultipleStringLiterals")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = { "sirius.web.test.enabled=" + OnEditingContextEventProcessorRegistryTests.KEY})
+public class EditingContextEventProcessorRegistryIntegrationTests extends AbstractIntegrationTests {
+
+    @Autowired
+    private IGivenInitialServerState givenInitialServerState;
+
+    @Autowired
+    private IGivenCommittedTransaction givenCommittedTransaction;
+
+    @Autowired
+    private IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    @Autowired
+    private IEditingContextSearchService editingContextSearchService;
+
+    @Autowired
+    private IObjectSearchService objectSearchService;
+
+    @BeforeEach
+    public void beforeEach() {
+        this.givenInitialServerState.initialize();
+    }
+
+    @Test
+    @GivenSiriusWebServer
+    @DisplayName("Given an editing context event processor registry, when we create an editing context event processor, then the IEditingContextEventProcessorEditingContextProcessors are executed.")
+    public void givenAnEditingContextEventProcessorRegistryWhenWeCreateAnEditingContextEventProcessorThenTheIEditingContextEventProcessorEditingContextProcessorsAreExecuted() {
+        this.givenCommittedTransaction.commit();
+
+        assertThat(this.editingContextEventProcessorRegistry.getEditingContextEventProcessors()).isEmpty();
+
+        // The creation of the EditingContextEventProcessor should also call all IEditingContextEventProcessorEditingContextProcessors
+        // In our case EcoreSampleEditingContextEventProcessorEditingContextProcessor will be called
+        // This processor only modifies the nsURI of the Sample ePackage
+        this.editingContextEventProcessorRegistry.getOrCreateEditingContextEventProcessor(TestIdentifiers.ECORE_SAMPLE_EDITING_CONTEXT_ID).orElse(null);
+
+        assertThat(this.editingContextEventProcessorRegistry.getEditingContextEventProcessors()).hasSize(1);
+
+        var optEditingContext = this.editingContextSearchService.findById(TestIdentifiers.ECORE_SAMPLE_EDITING_CONTEXT_ID);
+        assertTrue(optEditingContext.isPresent());
+
+        var optEPackage = this.objectSearchService.getObject(optEditingContext.get(), TestIdentifiers.EPACKAGE_OBJECT.toString());
+        assertTrue(optEPackage.isPresent());
+        assertTrue(optEPackage.get() instanceof EPackage);
+
+        EPackage ePackage = (EPackage) optEPackage.get();
+        assertEquals(EcoreSampleEditingContextEventProcessorEditingContextProcessor.NEW_NS_URI, ePackage.getNsURI());
```

Comment:
@sbegaudeau the test fails here and I don't understand why.
I tried to add `TestTransaction.flagCommit()/end()/start()` in different places of this test but it stills fail.
If you have any hints or If you want to update the test, please do no hesitate.